.github/workflows/release_notes.yml

```diff
@@ -7,13 +7,44 @@ on:
         description: 'Enter release PR number'
         required: true
         type: number
+  issue_comment:
+    types: [created]
+
+permissions:
+  contents: write
+  pull-requests: write
 
 jobs:
   release_notes:
     runs-on: ubuntu-latest
+    # Run only if dispatched or comment on a pull request
+    if: ${{ github.event_name == 'workflow_dispatch' || (github.event_name == 'issue_comment' && github.event.issue.pull_request && github.event.comment.body == 'run release_notes') }}
     steps:
+      # Determine if the triggering_actor is allowed to run this action
+      # We only permit admins
+      # Not only is 'triggering_actor' common between the trigger events it will also change if someone re-runs an old job
+      - name: check if triggering_actor is allowed to generate notes
+        env:
+          GH_TOKEN: ${{ github.token }}
+          COMMENTER: ${{ github.triggering_actor && github.triggering_actor || 'empty_triggering_actor' }}
+          API_ENDPOINT: /repos/${{ github.repository }}/collaborators?permission=admin
+        shell: bash
+        run: |
+          if [ $COMMENTER -eq "empty_triggering_actor" ]; then exit 1; fi
+          set -o pipefail
+          if gh api "$API_ENDPOINT" --paginate --jq ".[].login" | grep -q "$COMMENTER"; then
+            echo "$COMMENTER permitted to trigger notes!" && exit 0
+          else
+            echo "$COMMENTER not permitted to trigger notes" && exit 1
+          fi
+
+      # checkout the HEAD ref from prNumber
       - uses: actions/checkout@v3
+        with:
+          ref: refs/pull/${{ github.event_name == 'issue_comment' && github.event.issue.number || inputs.releasePr }}/head
```

Comment:
Why not store the PR number in the env variables since it is used here and in the code below to edit the pr body?

Reply:
What environment variables? There isn't a clean way to share variables btwn steps, we'd need a step to produce an output, I know it is a bit repetitive but an additional step is more complexity than the ternary is worth.

Reply:
Ah, good point. Was thinking of the `env` field under the `check if triggering_actor is allowed to generate notes` step, but I see that's localized to that step